File: Domain/Entities/Product.cs
using IPL.ECommerce.Domain.Enums;

namespace IPL.ECommerce.Domain.Entities;

public class Product
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public ProductType ProductType { get; set; }

    public decimal Price { get; set; }

    public int StockQuantity { get; set; }

    public string? ImageUrl { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public DateTime? ModifiedDate { get; set; }

    // Foreign Key
    public int FranchiseId { get; set; }

    // Navigation Property
    public Franchise Franchise { get; set; } = null!;
    public ICollection<CartItem> CartItems { get; set; }
        = new List<CartItem>();
    public ICollection<OrderItem> OrderItems { get; set; }
    = new List<OrderItem>();
}

File: ipl-ecommerce-ui/src/app/models/product.ts
export interface Product {

  id: number;

  name: string;

  description: string;

  productType: string;

  price: number;

  stockQuantity: number;

  imageUrl?: string;

  franchiseId: number;

  franchiseName: string;
}

export interface PagedResult<T> {

  items: T[];

  pageNumber: number;

  pageSize: number;

  totalCount: number;

  totalPages: number;

  hasPreviousPage: boolean;

  hasNextPage: boolean;
}